canonical-websites/www.ubuntu.com
# Fix feed timeout errors

Fixed #2109

We had an outage because the timeout was too large (60 seconds).

I have:

- Tidied up the handling of errors (so timeouts are handled in the same way as other errors)
- Reduced the default timeout to 10 seconds
- Extracted the feed work into [canonicalwebteam.get-feeds](https://github.com/canonical-webteam/get-feeds) for easy inclusion in other sites
- Switched www.ubuntu.com over to using the feed module

QA
--

`./run`

Now edit `templates/index.html` and replace the insights URL with something else (`http://example.com` for instance).

Check that `http://127.0.0.1:8001` loads the homepage, but shows an error where the feed should display.

If you want to test the timeout error specifically it's a bit harder. Ask Robin.
